src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -13,21 +13,55 @@ namespace Raven.Client.Documents.Commands.MultiGet
 {
     public class MultiGetCommand : RavenCommand<List<GetResponse>>
     {
+        private readonly RequestExecutor _requestExecutor;
         private readonly HttpCache _cache;
         private readonly List<GetRequest> _commands;
 
         private string _baseUrl;
 
-        public MultiGetCommand(HttpCache cache, List<GetRequest> commands)
+        public MultiGetCommand(RequestExecutor requestExecutor, List<GetRequest> commands)
         {
-            _cache = cache;
+            _requestExecutor = requestExecutor;
+            _cache = _requestExecutor.Cache;
             _commands = commands;
             ResponseType = RavenCommandResponseType.Raw;
         }
 
         public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
         {
             _baseUrl = $"{node.Url}/databases/{node.Database}";
+            url = $"{_baseUrl}/multi_get";
+
+            var aggressiveCacheOptions = _requestExecutor.AggressiveCaching.Value;
+            if (aggressiveCacheOptions?.Mode == AggressiveCacheMode.TrackChanges)
+            {
+                Result = new List<GetResponse>();
+                foreach (var command in _commands)
+                {
+                    if(command.SkipAggressiveCache) break;
```

Comment:
Should be `continue` ?

Reply:
No, the first one that isn't find for us, we'll break and the `Result`'s count will be different than `_commands` so we'll go to the server for all items.

Reply:
OK